fineract-provider/src/main/java/org/apache/fineract/portfolio/client/service/ClientReadPlatformServiceImpl.java

```diff
@@ -196,25 +197,20 @@ public ClientData retrieveTemplate(final Long officeId, final boolean staffInSel
         // this.context.validateAccessRights(searchParameters.getHierarchy());
         // underHierarchySearchString = searchParameters.getHierarchy() + "%";
         // }
-        List<Object> paramList = new ArrayList<>(Arrays.asList(underHierarchySearchString, underHierarchySearchString));
         final StringBuilder sqlBuilder = new StringBuilder(200);
         sqlBuilder.append("select SQL_CALC_FOUND_ROWS ");
         sqlBuilder.append(this.clientMapper.schema());
-        sqlBuilder.append(" where (o.hierarchy like ? or transferToOffice.hierarchy like ?) ");
+        final SQLBuilder extraCriteria = buildSqlStringFromClientCriteria(this.clientMapper.schema(), searchParameters);
+        sqlBuilder.append(" where (o.hierarchy like " + underHierarchySearchString + " or transferToOffice.hierarchy like "
+                + underHierarchySearchString + " ) ");
```

Comment:
@vorburger  can you please see this part? 
I am not sure how to handle this or condition in our SQL builder.